transport/zipkin/http.go

```diff
@@ -160,7 +162,17 @@ func (c *HTTPTransport) send(spans []*zipkincore.Span) error {
 		req.SetBasicAuth(c.httpCredentials.username, c.httpCredentials.password)
 	}
 
-	_, err = c.client.Do(req)
+	resp, err := c.client.Do(req)
+
+	if resp != nil && resp.StatusCode >= http.StatusBadRequest {
```

Comment:
need to check the error first. If there's no error then resp cannot be nil, so just check for stats < 400 and return nil, otherwise read the body.

Note that this code is largely duplicated with transport/http.go, I suggest defining a public function there that basically takes over from L155.